### PR TITLE
bazelisk.go: add simple .bazeliskrc file support

### PR DESCRIPTION
**Pull request by zellyn**
Replace `os.Getenv` with `getEnvOrConfig`, which falls back to parsing
key=value pairs from a `.bazeliskrc` file in the workspace
root. Environment variables take precedence.

Aims to solve #119 as simply as possible.

### CODE REVIEW

**Comment by zellyn** [comment]
fyi, @cheister

**Comment by joeljeske** [comment]
@philwo, is there anything that the community can do to help push this one through? I think this PR has a lot of merit, and would reduce configuration burden on newcomers to bazelisk.

**Comment by philwo** [comment]
@joeljeske Sorry, I've been super busy with finishing non-Bazelisk things that have to be done until end of this quarter - I'll get back to the issues and PRs here as soon as possible. Really looking forward to review / merge this and address any other open things.

File: bazelisk.go [hunk 1]
```diff
@@ -48,8 +49,53 @@ const (
...
+			log.Fatal(err)
+		}
+		fileConfig = make(map[string]string)
+		for _, line := range strings.Split(string(contents), "\n") {
+			parts := strings.SplitN(line, "=", 2)
+			if len(parts) < 2 {
+				continue
+			}
+			key := strings.TrimSpace(parts[0])
+			if strings.HasPrefix(key, "#") {
```
Comment (fweikert) [inline bazelisk.go:88]: Slightly more efficient if this check is the first line inside the loop

File: bazelisk.go [hunk 2]
```diff
@@ -48,8 +49,53 @@ const (
...
 )
 
+// getEnvOrConfig will read a configuration value from the environment, but fall back to reading it from .bazeliskrc in the workspace root.
+func getEnvOrConfig(name string) string {
+	if val := os.Getenv(name); val != "" {
+		return val
+	}
+
+	// Parse .bazeliskrc in the workspace root, once, if it can be found.
+	fileConfigOnce.Do(func() {
```
Comment (fweikert) [inline bazelisk.go:64]: Optional: for my taste a bit too long for an anonymous function

**Comment by philwo** [comment]
I talked with @fweikert, I'll just merge this now and we can polish later. 😊

**Pull request closed without merging**